src/custom/state/enhancedTransactions/updater.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useMemo } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { useActiveWeb3React } from 'hooks/web3'
+import { sdk } from 'utils/blocknative'
+import { cancelTransaction, replaceTransaction } from 'state/enhancedTransactions/actions'
+
+export default function Updater(): null {
+  const { chainId, library } = useActiveWeb3React()
+  const dispatch = useAppDispatch()
+  const state = useAppSelector((state) => state.transactions)
+
+  const transactions = useMemo(() => (chainId ? state[chainId] ?? {} : {}), [chainId, state])
+
+  useEffect(() => {
+    if (!chainId || !library) return
+
+    const pendingHashes = Object.keys(transactions).filter((hash) => !transactions[hash].receipt)
```

Comment:
using premade hook for this here: `L48` inside `src/state/transactions/hooks.tsx`